File: backend/app/schemas/assistant.py
from __future__ import annotations

from pydantic import BaseModel, Field

NO_CONTEXT_ANSWER = (
    "Je n'ai pas trouvé suffisamment d'informations dans les documents "
    "sélectionnés pour répondre à cette question."
)


class AskQuestionRequest(BaseModel):
    document_ids: list[int] = Field(min_length=1)
    question: str = Field(min_length=1, max_length=2000)


class SourceCitation(BaseModel):
    document_id: int
    document_title: str
    page_number: int = Field(ge=1)
    excerpt: str = Field(min_length=1)


class KnowledgeAnswer(BaseModel):
    answer: str
    has_sufficient_context: bool
    sources: list[SourceCitation] = Field(default_factory=list)


class AskQuestionResponse(KnowledgeAnswer):
    pass

File: frontend/src/features/assistant/types/assistant.ts
export type SourceCitation = {
  documentId: number;
  documentTitle: string;
  pageNumber: number;
  excerpt: string;
};

export type AskQuestionRequest = {
  question: string;
  documentIds: number[];
};

export type AskQuestionResponse = {
  answer: string;
  hasSufficientContext: boolean;
  sources: SourceCitation[];
};

export type AssistantMessage = {
  id: string;
  role: "assistant" | "user";
  content: string;
  sources?: SourceCitation[];
};
